internal/engine/experiment/torsf/torsf.go

```diff
@@ -101,47 +127,127 @@ func (m *Measurer) Run(
+	}()
 	tun, err := m.startTunnel()(ctx, &tunnel.Config{
 		Name:      "tor",
 		Session:   sess,
-		TunnelDir: path.Join(sess.TempDir(), "torsf"),
+		TunnelDir: path.Join(m.baseTunnelDir(sess), "torsf"),
 		Logger:    sess.Logger(),
 		TorArgs: []string{
 			"UseBridges", "1",
 			"ClientTransportPlugin", ptl.AsClientTransportPluginArgument(),
 			"Bridge", sfdialer.AsBridgeArgument(),
 		},
 	})
+	m.readTorLogs(sess.Logger(), tk, tun)
 	if err != nil {
 		// Note: archival.NewFailure scrubs IP addresses
-		testkeys.Failure = archival.NewFailure(err)
-		// This error condition means we could not bootstrap with snowflake
-		// for $reasons, so the experiment didn't fail, rather it did record
-		// that something prevented snowflake from running.
-		errch <- nil
+		tk.Failure = archival.NewFailure(err)
 		return
 	}
 	defer tun.Stop()
-	testkeys.BootstrapTime = tun.BootstrapTime().Seconds()
-	errch <- nil
+	tk.BootstrapTime = tun.BootstrapTime().Seconds()
+}
+
+// readTorLogs attempts to read and include the tor logs into
+// the test keys if this operation is possible.
+//
+// This function aims to _only_ include:
+//
+// 1. information on the tor version by intercepting the line that
+// writes which tor version is opening a log file;
+//
+// 2. notices (more detailed debug messages may contain information
+// that we'd rather not include into the logs?);
+//
+// 3. information about bootstrap (by looking at the progress of
+// the bootstrap we understand where it blocks and we also know the
+// amount of work tor needs to do, hence we know the cache status
+// because a working cache includes much less messages);
+//
+// 4. information about bridges being used (from there we know
+// if the bridge was cached of fresh, by the way).
+//
+// Tor is know to be good software that does not break its output
+// unnecessarily and that does not include PII into its logs unless
+// explicitly asked to. This fact gives me confidence that we can
+// safely include this subset of the logs into the results.
+//
+// On this note, I think it's safe to include timestamps from the
+// logs into the output, since we have a timestamp for the whole
+// experiment already, so we don't leak much more by also including
+// the Tor proper timestamps into the results.
+func (m *Measurer) readTorLogs(logger model.Logger, tk *TestKeys, tun tunnel.Tunnel) {
+	logFilePath, found := tun.LogFilePath()
+	if !found {
+		log.Warn("the tunnel claims it contains no log")
+		return
+	}
+	data, err := os.ReadFile(logFilePath)
+	if err != nil {
+		log.Warnf("could not read tor logs: %s", err.Error())
+		return
+	}
+	for _, bline := range bytes.Split(data, []byte("\n")) {
+		sline := string(bline) // avoid IP addresses in logs
+		if strings.HasSuffix(sline, "opening a new log file.") {
+			tk.TorLogs = append(tk.TorLogs, sline)
+		}
+		if !strings.Contains(sline, "[notice]") {
+			continue
+		}
+		sline = scrubber.Scrub(sline) // maybe unneeded paranoia to avoid including IPs
```

Comment:
It looks like this scrubber is a bit too paranoid. When opening it up to log all messages, it's also matching the Tor version string which it thinks is an ipv4 address (ex. `0.4.6.7`).

Since we are already performing the stripping of the probe_ip at the JSON encoding level, shouldn't we be OK removing this?

Reply:
Okay, I removed the scrubber